### PR TITLE
fix(spanner): run system test against emulator skip not supported tests by emulator

### PR DESCRIPTION
**Pull request by jiren**
closes: #5936 

export `SPANNER_EMULATOR_HOST` env variable to enable system tests runs against emulator.

### CODE REVIEW

File: google-cloud-spanner/acceptance/spanner_helper.rb [hunk 1]
```diff
@@ -31,8 +31,16 @@ def SecureRandom.int64
...
+  ENV["SPANNER_EMULATOR_HOST"]
+end
+
+def emulator_enabled?
+  !emulator_host.nil?
+end
+
 # Create shared spanner object so we don't create new for each test
-$spanner = Google::Cloud::Spanner.new
+$spanner = Google::Cloud::Spanner.new(emulator_host: emulator_host)
```
Comment (hengfengli) [inline google-cloud-spanner/acceptance/spanner_helper.rb:43]: If `SPANNER_EMULATOR_HOST` is set, we should not need to pass it as a parameter: 

https://github.com/googleapis/google-cloud-ruby/blob/5f61bf15a744f8d969e6bdae80f57ac8ea150d5b/google-cloud-spanner/test/google/cloud/spanner_test.rb#L325-L341

File: google-cloud-spanner/Rakefile [hunk 2]
```diff
@@ -98,7 +100,7 @@ namespace :acceptance do
     $LOAD_PATH.unshift "lib"
     require "google/cloud/spanner"
     puts "Cleaning up Spanner instances and databases."
-    Google::Cloud::Spanner.new.instances.all do |instance|
+    Google::Cloud::Spanner.new(emulator_host: emulator_host).instances.all do |instance|
```
Comment (hengfengli) [inline google-cloud-spanner/Rakefile:103]: Same here. If `SPANNER_EMULATOR_HOST` is set, we should not need to pass it as a parameter: 

https://github.com/googleapis/google-cloud-ruby/blob/5f61bf15a744f8d969e6bdae80f57ac8ea150d5b/google-cloud-spanner/test/google/cloud/spanner_test.rb#L325-L341

File: google-cloud-spanner/Rakefile [hunk 3]
```diff
@@ -85,6 +85,8 @@ namespace :acceptance do
     if project.nil? || keyfile.nil?
       fail "You must provide a project and keyfile. e.g. rake acceptance[test123, /path/to/keyfile.json] or SPANNER_TEST_PROJECT=test123 SPANNER_TEST_KEYFILE=/path/to/keyfile.json rake acceptance"
     end
+    emulator_host = ENV["SPANNER_EMULATOR_HOST"]
```
Comment (hengfengli) [inline google-cloud-spanner/Rakefile:88]: This can be deleted.

File: google-cloud-spanner/acceptance/spanner_helper.rb (hunk 1 above)
Comment (jiren) [reply google-cloud-spanner/acceptance/spanner_helper.rb]: done

File: google-cloud-spanner/Rakefile (hunk 2 above)
Comment (jiren) [reply google-cloud-spanner/Rakefile]: done

File: google-cloud-spanner/Rakefile (hunk 3 above)
Comment (jiren) [reply google-cloud-spanner/Rakefile]: done

File: google-cloud-spanner/acceptance/spanner_helper.rb [hunk 4]
```diff
@@ -31,16 +31,12 @@ def SecureRandom.int64
   random_bytes(8).unpack("q")[0]
 end
 
-def emulator_host
-  ENV["SPANNER_EMULATOR_HOST"]
-end
-
 def emulator_enabled?
-  !emulator_host.nil?
+  !ENV["SPANNER_EMULATOR_HOST"].nil?
```
Comment (hengfengli) [inline google-cloud-spanner/acceptance/spanner_helper.rb:35]: I think can just use 

```ruby
def emulator_enabled?
  ENV["SPANNER_EMULATOR_HOST"]
end
```

https://stackoverflow.com/questions/4012775/is-there-a-ruby-or-ruby-ism-for-not-nil-opposite-of-nil-method

Comment (jiren) [reply google-cloud-spanner/acceptance/spanner_helper.rb]: done

File: google-cloud-spanner/Rakefile [hunk 5]
```diff
@@ -85,6 +85,7 @@ namespace :acceptance do
     if project.nil? || keyfile.nil?
       fail "You must provide a project and keyfile. e.g. rake acceptance[test123, /path/to/keyfile.json] or SPANNER_TEST_PROJECT=test123 SPANNER_TEST_KEYFILE=/path/to/keyfile.json rake acceptance"
     end
+
```
Comment (hengfengli) [inline google-cloud-spanner/Rakefile:88]: Please remove this empty line.

**Pull request merged**